app/portainer/services/endpointProvider.js

```diff
@@ -37,5 +52,33 @@ angular.module('portainer.app')
     LocalStorage.storeEndpointPublicURL(publicURL);
   };
 
+  service.endpoints = function() {
+    return LocalStorage.getEndpoints();
+  };
+
+  service.setEndpoints = function(data) {
+    LocalStorage.storeEndpoints(data);
+  };
+
+  service.endpointStatus = function() {
+    if (endpoint.Status === undefined) {
+      endpoint.Status = LocalStorage.getEndpointStatus();
+    }
+    return endpoint.Status;
+  };
+
+  service.setEndpointStatus = function(status) {
+    endpoint.Status = status;
+    LocalStorage.storeEndpointStatus(status);
+  };
+
+  service.currentEndpoint = function() {
```

Comment:
This whole section seems like a big workaround for me but it will be ok for now. Ultimately we need to think about a way to store the entire endpoint object instead of storing properties like id and public URL.

Reply:
Maybe use JSON.parse/stringify?